src/ripple/rpc/impl/KeypairForSignature.h

```diff
@@ -33,6 +34,7 @@ struct KeyPair
 };
 
 KeyPair keypairForSignature (Json::Value const& params, Json::Value& error);
```

Comment:
Your example of `std::tie(pk, sk)` still offers no compile-time protection (and is in fact backwards).

Since RippleAddress is going to be replaced anyway, I don't see the big deal of removing KeyPair at the same time.